demo/login.cgi

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/python
+r"""
+Taken from https://github.com/plone/plone.session/blob/master/plone/session/tktauth.py
+It needed some modifications to work properly:
+- The IP must be encoded as a string (plone was using 4-bytes). The reason mod_authn_tkt
+does this is because an IP address might be IPv4 or it might be IPv6. An IPv6 address
+does not fit into 4 packed bytes (which is the size of an IPv4 address)
+- The first digest (digest0) used to calculate the final digest in mod_auth_tkt_digest()
+must be the 16-bytes array while the original used the hex-string
+"""
+
+from struct import pack
+
+import hashlib
+import hmac
+import six
+import time
+import sys
+import os
+
+def safe_encode(value, encoding='utf-8'):
+    """Convert unicode to the specified encoding.
+
+    copied from Products.CMFPlone.utils b/c this package does not depend on it
+    """
+    if isinstance(value, six.text_type):
+        value = value.encode(encoding)
+    return value
+
+def mod_auth_tkt_digest(secret, data1, data2):
+    # data1: addr + timestamp
+    # data2: userid+token_list+user_data
+    #digest0 = hashlib.md5(data1 + secret + data2).hexdigest() # THIS line that was modified to work with lighttpd mod_authn_tkt
+    # The original code was using the hex-string while mod_authn_tkt uses the bytes
+    digest0 = hashlib.md5(data1 + secret + data2).digest()
+    if not six.PY2:
+        # In Python 3 hashlib.md5(value).hexdigest() wants a bites value
```

Comment:
s/bites/bytes/

Reply:
apparently `bites` is also from https://github.com/plone/plone.session/blob/master/plone/session/tktauth.py